external/loaders/loaders/batches/_sequences.py

```diff
@@ -69,13 +69,20 @@ def files(self):
 
     @classmethod
     def dump(cls, dataset, path):
-        joblib.dump(dataset, path)
+        try:
+            joblib.dump(dataset.load(), path)
+        except AttributeError:
+            joblib.dump(dataset, path)
```

Comment:
Duck-typing is fine, but I don't think you want to catch attribute errors which may occur in joblib.dump
```suggestion
        try:
            loaded_data = dataset.load()
        except AttributeError:
            pass
        else:
            loaded_data = dataset
         joblib.dump(loaded_data, path)
```